server/plugin/api.go

```diff
@@ -869,7 +870,78 @@ func (p *Plugin) handleSubscriptionNotifications(w http.ResponseWriter, r *http.
 			},
 		}
 	case constants.SubscriptionEventRunStageWaitingForApproval:
-		// TODO
+		organization := ""
+		webLinkPaths := strings.Split(body.Resource.Release.ReleaseDefinition.Links.Web.Href, "/")
+		if len(webLinkPaths) >= 4 {
+			organization = webLinkPaths[3]
+		}
+
+		approverTitle := "Approver(s)"
+		if body.Resource.Approval.ExecutionOrder == "inSequence" {
+			approverTitle = "Approver(s) in sequence"
+		} else if body.Resource.Approval.MinRequiredApprovers > 0 && len(body.Resource.Approval.Steps) > body.Resource.Approval.MinRequiredApprovers {
+			approverTitle = fmt.Sprintf("Approvers (any %d)", body.Resource.Approval.MinRequiredApprovers)
+		}
+
+		approvers := ""
+		for _, approvalStep := range body.Resource.Approval.Steps {
+			approvers = approvers + approvalStep.AssignedApprover.DisplayName + "\n"
+		}
+
+		attachment = &model.SlackAttachment{
+			Pretext:    body.Message.Markdown,
+			AuthorName: constants.SlackAttachmentAuthorNamePipelines,
+			AuthorIcon: fmt.Sprintf(constants.StaticFiles, p.GetSiteURL(), constants.PluginID, constants.FileNamePipelinesIcon),
+			Color:      constants.IconColorPipelines,
+			Fields: []*model.SlackAttachmentField{
+				{
+					Title: "Run pipeline",
+					Value: fmt.Sprintf("[%s](%s)", body.Resource.Pipeline.Name, body.Resource.Pipeline.Links.Web.Href),
+					Short: true,
+				},
+				{
+					Title: "Stage",
+					Value: fmt.Sprintf("[%s](%s)", body.Resource.Stage.Name, body.Resource.Stage.Links.Web.Href),
+					Short: true,
+				},
+				{
+					Title: approverTitle,
+					Value: approvers,
+				},
+			},
+			Actions: []*model.PostAction{
+				{
+					Id:    constants.PipelineRequestIDApproved,
+					Type:  "button",
```

Comment:
```suggestion
					Type:  model.POST_ACTION_TYPE_BUTTON,
```
Please fix this in the whole repo.